### PR TITLE
Fluent : carousel accessibility fixes

### PR DESCRIPTION
**Pull request by kolaps33**
This PR is created based on the from another repository:
https://github.com/microsoft/fluent-ui-react/pull/2278/files

Related carousel: Carousel with Pagination

In this PR are fixes related to the feedback which we got from Rekha + feedback from testing.

1. 
Issue: Index numbers (1 of 4) are read twice as you down arrow through the carousel item.
Fix: add aria-hidden="true" on the text

2.1 Issue: The first and the last item in the carousel are not read when the prev/next button is pressed in browse mode.
2.2 #2224
Fix: moving focusing the next/previous paddle to generic function, which is call always, even when "onclick" is executed...

3.
Issue: Should there be focus on the entire carousel control and the left/right buttons? Seems redundant and adds more keyboard stops
Fix: change the behavior that for carousel without the "tab" navigation there will be tabindex=-1 on itemsContainer

4.
Issue: Carousel itself should have a label property defined. The roledescription is set to “Carousel” so label can be “Portrait collection” for this example
Fix: added prop aria-label to be able label the carousel

5.
Issue: #2225
Fix: was decided that if reader will narrate "carousel" user will know how to use keyboard to rotate through the carousel
For this reason to narrate "carousel" in application mode was added additional props(role, aria-label,aria-roledescription) into "itemsContainer" in "carouselBehavior.ts".
Now there is role "group". After more testing could be change for "region". For me region was not narrating "carousel" in focus mode for NVDA.

6.
Issue: Programmatically Prev/Next buttons are outside carousel container so it will likely confuse the user. Carousel container element should include all elements within carousel.
Fix: apply region only for "Carousel with Pagination", provide there proper label and roledescription. Fix is in "carouselBehavior.ts" for "root" slot.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/12050)

### CODE REVIEW

**Comment by JasonGore** [comment]
This PR looks like it succumbed to github issues that were occurring last night so I will close and reopen to trigger a new build.

**Comment by aneeshack4** [comment]
@kolaps33 Any updates on this?

File: packages/fluentui/docs/src/examples/components/Carousel/BestPractices/CarouselBestPractices.tsx [hunk 1]
```diff
@@ -8,7 +8,16 @@ const doList = [
...
     'Add textual representation for `CarouselItem`. Use `aria-label` attribute ( refer to{' '}
     {link('reported issue', 'https://bugs.chromium.org/p/chromium/issues/detail?id=1040924')} for details).
-  </Text>
+  </Text>,
+  'Provide  localized string of the "carousel" using `ariaRoleDescription` prop.',
+  'Provide label to the carousel using `ariaLabel` prop.',
+  <Box>
+    If carousel contains `navigation`:
+    <ul>
+      <li> provide label to `navigation` and to navigation item using `aria-label` attribute</li>
```
Comment (jurokapsiar) [inline packages/fluentui/docs/src/examples/components/Carousel/BestPractices/CarouselBestPractices.tsx]: provide label to `navigation` slot?

File: packages/fluentui/docs/src/examples/components/Carousel/BestPractices/CarouselBestPractices.tsx [hunk 2]
```diff
@@ -8,7 +8,16 @@ const doList = [
...
+      <li> add `aria-controls` attribute to navigation item referencing to `carouselItem` id </li>
```
Comment (jurokapsiar) [inline packages/fluentui/docs/src/examples/components/Carousel/BestPractices/CarouselBestPractices.tsx]: do we have an example that uses `aria-controls`?

Comment (kolaps33) [reply packages/fluentui/docs/src/examples/components/Carousel/BestPractices/CarouselBestPractices.tsx]: yes, already first Carousel example on the page 
https://fluentsite.z22.web.core.windows.net/components/carousel/definition
using 'aria-controls'

File: packages/fluentui/docs/src/examples/components/Carousel/BestPractices/CarouselBestPractices.tsx (hunk 1 above)
Comment (kolaps33) [reply packages/fluentui/docs/src/examples/components/Carousel/BestPractices/CarouselBestPractices.tsx]: change to "navigation slot"

**Pull request merged**